### ISSUE TITLE
Installation has failed (Windows 10) 

### ISSUE THREAD

**Issue by SHENKAR** [atom/atom#10383]
2016-01-11 21:40:07> Program: Starting Squirrel Updater: --install .
2016-01-11 21:40:07> Program: Starting install, writing to C:\Users\mypc\AppData\Local\SquirrelTemp
2016-01-11 21:40:07> Program: About to install to: C:\Users\mypc\AppData\Local\atom
2016-01-11 21:40:07> CheckForUpdateImpl: Failed to load local releases, starting from scratch: System.IO.DirectoryNotFoundException: Could not find a part of the path 'C:\Users\mypc\AppData\Local\atom\packages\RELEASES'.
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share)
   at Squirrel.Utility.LoadLocalReleases(String localReleaseFile)
   at Squirrel.UpdateManager.CheckForUpdateImpl.<CheckForUpdate>d__3d.MoveNext()
2016-01-11 21:40:07> CheckForUpdateImpl: Reading RELEASES file from C:\Users\mypc\AppData\Local\SquirrelTemp
2016-01-11 21:40:07> CheckForUpdateImpl: First run or local directory is corrupt, starting from scratch
2016-01-11 21:40:07> ApplyReleasesImpl: Writing files to app directory: C:\Users\mypc\AppData\Local\atom\app-1.3.3
2016-01-11 21:40:29> IEnableLogger: Failed to install package to app dir: System.AggregateException: One or more errors occurred. ---> System.IO.IOException: Access to the path 'resources' is denied.
   at System.IO.DirectoryInfo.MoveTo(String destDirName)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClassaf.<>c__DisplayClassb4.<>c__DisplayClassb6.<installPackageToAppDir>b__ad(DirectoryInfo subdir)
   at Squirrel.Utility.<>c__DisplayClass18`1.<>c__DisplayClass1a.<ForEachAsync>b__17()
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.Utility.<>c__DisplayClass1e`1.<>c__DisplayClass20.<<ForEachAsync>b__1d>d__22.MoveNext()
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClassaf.<>c__DisplayClassb4.<installPackageToAppDir>b__ac(DirectoryInfo ld)
   at Squirrel.EnumerableExtensions.ForEach[TSource](IEnumerable`1 source, Action`1 onNext)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClassaf.<<installPackageToAppDir>b__a8>d__b8.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.Utility.<LogIfThrows>d__54`1.MoveNext()
---> (Inner Exception #0) System.IO.IOException: Access to the path 'resources' is denied.
   at System.IO.DirectoryInfo.MoveTo(String destDirName)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClassaf.<>c__DisplayClassb4.<>c__DisplayClassb6.<installPackageToAppDir>b__ad(DirectoryInfo subdir)
   at Squirrel.Utility.<>c__DisplayClass18`1.<>c__DisplayClass1a.<ForEachAsync>b__17()
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.Utility.<>c__DisplayClass1e`1.<>c__DisplayClass20.<<ForEachAsync>b__1d>d__22.MoveNext()<---

2016-01-11 21:40:29> Unhandled exception: System.AggregateException: One or more errors occurred. ---> System.AggregateException: One or more errors occurred. ---> System.IO.IOException: Access to the path 'resources' is denied.
   at System.IO.DirectoryInfo.MoveTo(String destDirName)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClassaf.<>c__DisplayClassb4.<>c__DisplayClassb6.<installPackageToAppDir>b__ad(DirectoryInfo subdir)
   at Squirrel.Utility.<>c__DisplayClass18`1.<>c__DisplayClass1a.<ForEachAsync>b__17()
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.Utility.<>c__DisplayClass1e`1.<>c__DisplayClass20.<<ForEachAsync>b__1d>d__22.MoveNext()
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClassaf.<>c__DisplayClassb4.<installPackageToAppDir>b__ac(DirectoryInfo ld)
   at Squirrel.EnumerableExtensions.ForEach[TSource](IEnumerable`1 source, Action`1 onNext)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClassaf.<<installPackageToAppDir>b__a8>d__b8.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.Utility.<LogIfThrows>d__54`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<ApplyReleases>d__78.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.UpdateManager.<FullInstall>d__f.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at Squirrel.Update.Program.<Install>d__37.MoveNext()
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait()
   at Squirrel.Update.Program.executeCommandLine(String[] args)
   at Squirrel.Update.Program.main(String[] args)
---> (Inner Exception #0) System.AggregateException: One or more errors occurred. ---> System.IO.IOException: Access to the path 'resources' is denied.
   at System.IO.DirectoryInfo.MoveTo(String destDirName)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClassaf.<>c__DisplayClassb4.<>c__DisplayClassb6.<installPackageToAppDir>b__ad(DirectoryInfo subdir)
   at Squirrel.Utility.<>c__DisplayClass18`1.<>c__DisplayClass1a.<ForEachAsync>b__17()
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.Utility.<>c__DisplayClass1e`1.<>c__DisplayClass20.<<ForEachAsync>b__1d>d__22.MoveNext()
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClassaf.<>c__DisplayClassb4.<installPackageToAppDir>b__ac(DirectoryInfo ld)
   at Squirrel.EnumerableExtensions.ForEach[TSource](IEnumerable`1 source, Action`1 onNext)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClassaf.<<installPackageToAppDir>b__a8>d__b8.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.Utility.<LogIfThrows>d__54`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<ApplyReleases>d__78.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.UpdateManager.<FullInstall>d__f.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at Squirrel.Update.Program.<Install>d__37.MoveNext()
---> (Inner Exception #0) System.IO.IOException: Access to the path 'resources' is denied.
   at System.IO.DirectoryInfo.MoveTo(String destDirName)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClassaf.<>c__DisplayClassb4.<>c__DisplayClassb6.<installPackageToAppDir>b__ad(DirectoryInfo subdir)
   at Squirrel.Utility.<>c__DisplayClass18`1.<>c__DisplayClass1a.<ForEachAsync>b__17()
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.Utility.<>c__DisplayClass1e`1.<>c__DisplayClass20.<<ForEachAsync>b__1d>d__22.MoveNext()<---
<---

2016-01-11 22:05:58> Program: Starting Squirrel Updater: --install .
2016-01-11 22:05:58> Program: Starting install, writing to C:\Users\mypc\AppData\Local\SquirrelTemp
2016-01-11 22:05:58> Program: About to install to: C:\Users\mypc\AppData\Local\atom
2016-01-11 22:05:58> Program: Install path C:\Users\mypc\AppData\Local\atom already exists, burning it to the ground
2016-01-11 22:05:59> CheckForUpdateImpl: Failed to load local releases, starting from scratch: System.IO.DirectoryNotFoundException: Could not find a part of the path 'C:\Users\mypc\AppData\Local\atom\packages\RELEASES'.
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share)
   at Squirrel.Utility.LoadLocalReleases(String localReleaseFile)
   at Squirrel.UpdateManager.CheckForUpdateImpl.<CheckForUpdate>d__3d.MoveNext()
2016-01-11 22:05:59> CheckForUpdateImpl: Reading RELEASES file from C:\Users\mypc\AppData\Local\SquirrelTemp
2016-01-11 22:05:59> CheckForUpdateImpl: First run or local directory is corrupt, starting from scratch
2016-01-11 22:06:00> ApplyReleasesImpl: Writing files to app directory: C:\Users\mypc\AppData\Local\atom\app-1.3.3
2016-01-11 22:06:20> IEnableLogger: Failed to install package to app dir: System.AggregateException: One or more errors occurred. ---> System.IO.IOException: Access to the path 'resources' is denied.
   at System.IO.DirectoryInfo.MoveTo(String destDirName)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClassaf.<>c__DisplayClassb4.<>c__DisplayClassb6.<installPackageToAppDir>b__ad(DirectoryInfo subdir)
   at Squirrel.Utility.<>c__DisplayClass18`1.<>c__DisplayClass1a.<ForEachAsync>b__17()
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.Utility.<>c__DisplayClass1e`1.<>c__DisplayClass20.<<ForEachAsync>b__1d>d__22.MoveNext()
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClassaf.<>c__DisplayClassb4.<installPackageToAppDir>b__ac(DirectoryInfo ld)
   at Squirrel.EnumerableExtensions.ForEach[TSource](IEnumerable`1 source, Action`1 onNext)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClassaf.<<installPackageToAppDir>b__a8>d__b8.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.Utility.<LogIfThrows>d__54`1.MoveNext()
---> (Inner Exception #0) System.IO.IOException: Access to the path 'resources' is denied.
   at System.IO.DirectoryInfo.MoveTo(String destDirName)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClassaf.<>c__DisplayClassb4.<>c__DisplayClassb6.<installPackageToAppDir>b__ad(DirectoryInfo subdir)
   at Squirrel.Utility.<>c__DisplayClass18`1.<>c__DisplayClass1a.<ForEachAsync>b__17()
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.Utility.<>c__DisplayClass1e`1.<>c__DisplayClass20.<<ForEachAsync>b__1d>d__22.MoveNext()<---

2016-01-11 22:06:20> Unhandled exception: System.AggregateException: One or more errors occurred. ---> System.AggregateException: One or more errors occurred. ---> System.IO.IOException: Access to the path 'resources' is denied.
   at System.IO.DirectoryInfo.MoveTo(String destDirName)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClassaf.<>c__DisplayClassb4.<>c__DisplayClassb6.<installPackageToAppDir>b__ad(DirectoryInfo subdir)
   at Squirrel.Utility.<>c__DisplayClass18`1.<>c__DisplayClass1a.<ForEachAsync>b__17()
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.Utility.<>c__DisplayClass1e`1.<>c__DisplayClass20.<<ForEachAsync>b__1d>d__22.MoveNext()
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClassaf.<>c__DisplayClassb4.<installPackageToAppDir>b__ac(DirectoryInfo ld)
   at Squirrel.EnumerableExtensions.ForEach[TSource](IEnumerable`1 source, Action`1 onNext)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClassaf.<<installPackageToAppDir>b__a8>d__b8.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.Utility.<LogIfThrows>d__54`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<ApplyReleases>d__78.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.UpdateManager.<FullInstall>d__f.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at Squirrel.Update.Program.<Install>d__37.MoveNext()
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait()
   at Squirrel.Update.Program.executeCommandLine(String[] args)
   at Squirrel.Update.Program.main(String[] args)
---> (Inner Exception #0) System.AggregateException: One or more errors occurred. ---> System.IO.IOException: Access to the path 'resources' is denied.
   at System.IO.DirectoryInfo.MoveTo(String destDirName)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClassaf.<>c__DisplayClassb4.<>c__DisplayClassb6.<installPackageToAppDir>b__ad(DirectoryInfo subdir)
   at Squirrel.Utility.<>c__DisplayClass18`1.<>c__DisplayClass1a.<ForEachAsync>b__17()
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.Utility.<>c__DisplayClass1e`1.<>c__DisplayClass20.<<ForEachAsync>b__1d>d__22.MoveNext()
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClassaf.<>c__DisplayClassb4.<installPackageToAppDir>b__ac(DirectoryInfo ld)
   at Squirrel.EnumerableExtensions.ForEach[TSource](IEnumerable`1 source, Action`1 onNext)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClassaf.<<installPackageToAppDir>b__a8>d__b8.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.Utility.<LogIfThrows>d__54`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<ApplyReleases>d__78.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.UpdateManager.<FullInstall>d__f.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at Squirrel.Update.Program.<Install>d__37.MoveNext()
---> (Inner Exception #0) System.IO.IOException: Access to the path 'resources' is denied.
   at System.IO.DirectoryInfo.MoveTo(String destDirName)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClassaf.<>c__DisplayClassb4.<>c__DisplayClassb6.<installPackageToAppDir>b__ad(DirectoryInfo subdir)
   at Squirrel.Utility.<>c__DisplayClass18`1.<>c__DisplayClass1a.<ForEachAsync>b__17()
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.Utility.<>c__DisplayClass1e`1.<>c__DisplayClass20.<<ForEachAsync>b__1d>d__22.MoveNext()<---
<---

2016-01-11 22:06:48> Program: Starting Squirrel Updater: --install .
2016-01-11 22:06:48> Program: Starting install, writing to C:\Users\mypc\AppData\Local\SquirrelTemp
2016-01-11 22:06:48> Program: About to install to: C:\Users\mypc\AppData\Local\atom
2016-01-11 22:06:48> Program: Install path C:\Users\mypc\AppData\Local\atom already exists, burning it to the ground
2016-01-11 22:06:49> CheckForUpdateImpl: Failed to load local releases, starting from scratch: System.IO.DirectoryNotFoundException: Could not find a part of the path 'C:\Users\mypc\AppData\Local\atom\packages\RELEASES'.
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share)
   at Squirrel.Utility.LoadLocalReleases(String localReleaseFile)
   at Squirrel.UpdateManager.CheckForUpdateImpl.<CheckForUpdate>d__3d.MoveNext()
2016-01-11 22:06:49> CheckForUpdateImpl: Reading RELEASES file from C:\Users\mypc\AppData\Local\SquirrelTemp
2016-01-11 22:06:49> CheckForUpdateImpl: First run or local directory is corrupt, starting from scratch
2016-01-11 22:06:49> ApplyReleasesImpl: Writing files to app directory: C:\Users\mypc\AppData\Local\atom\app-1.3.3
2016-01-11 22:07:11> IEnableLogger: Failed to install package to app dir: System.AggregateException: One or more errors occurred. ---> System.IO.IOException: Access to the path 'resources' is denied.
   at System.IO.DirectoryInfo.MoveTo(String destDirName)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClassaf.<>c__DisplayClassb4.<>c__DisplayClassb6.<installPackageToAppDir>b__ad(DirectoryInfo subdir)
   at Squirrel.Utility.<>c__DisplayClass18`1.<>c__DisplayClass1a.<ForEachAsync>b__17()
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.Utility.<>c__DisplayClass1e`1.<>c__DisplayClass20.<<ForEachAsync>b__1d>d__22.MoveNext()
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClassaf.<>c__DisplayClassb4.<installPackageToAppDir>b__ac(DirectoryInfo ld)
   at Squirrel.EnumerableExtensions.ForEach[TSource](IEnumerable`1 source, Action`1 onNext)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClassaf.<<installPackageToAppDir>b__a8>d__b8.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.Utility.<LogIfThrows>d__54`1.MoveNext()
---> (Inner Exception #0) System.IO.IOException: Access to the path 'resources' is denied.
   at System.IO.DirectoryInfo.MoveTo(String destDirName)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClassaf.<>c__DisplayClassb4.<>c__DisplayClassb6.<installPackageToAppDir>b__ad(DirectoryInfo subdir)
   at Squirrel.Utility.<>c__DisplayClass18`1.<>c__DisplayClass1a.<ForEachAsync>b__17()
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.Utility.<>c__DisplayClass1e`1.<>c__DisplayClass20.<<ForEachAsync>b__1d>d__22.MoveNext()<---

2016-01-11 22:07:11> Unhandled exception: System.AggregateException: One or more errors occurred. ---> System.AggregateException: One or more errors occurred. ---> System.IO.IOException: Access to the path 'resources' is denied.
   at System.IO.DirectoryInfo.MoveTo(String destDirName)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClassaf.<>c__DisplayClassb4.<>c__DisplayClassb6.<installPackageToAppDir>b__ad(DirectoryInfo subdir)
   at Squirrel.Utility.<>c__DisplayClass18`1.<>c__DisplayClass1a.<ForEachAsync>b__17()
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.Utility.<>c__DisplayClass1e`1.<>c__DisplayClass20.<<ForEachAsync>b__1d>d__22.MoveNext()
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClassaf.<>c__DisplayClassb4.<installPackageToAppDir>b__ac(DirectoryInfo ld)
   at Squirrel.EnumerableExtensions.ForEach[TSource](IEnumerable`1 source, Action`1 onNext)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClassaf.<<installPackageToAppDir>b__a8>d__b8.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.Utility.<LogIfThrows>d__54`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<ApplyReleases>d__78.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.UpdateManager.<FullInstall>d__f.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at Squirrel.Update.Program.<Install>d__37.MoveNext()
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait()
   at Squirrel.Update.Program.executeCommandLine(String[] args)
   at Squirrel.Update.Program.main(String[] args)
---> (Inner Exception #0) System.AggregateException: One or more errors occurred. ---> System.IO.IOException: Access to the path 'resources' is denied.
   at System.IO.DirectoryInfo.MoveTo(String destDirName)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClassaf.<>c__DisplayClassb4.<>c__DisplayClassb6.<installPackageToAppDir>b__ad(DirectoryInfo subdir)
   at Squirrel.Utility.<>c__DisplayClass18`1.<>c__DisplayClass1a.<ForEachAsync>b__17()
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.Utility.<>c__DisplayClass1e`1.<>c__DisplayClass20.<<ForEachAsync>b__1d>d__22.MoveNext()
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClassaf.<>c__DisplayClassb4.<installPackageToAppDir>b__ac(DirectoryInfo ld)
   at Squirrel.EnumerableExtensions.ForEach[TSource](IEnumerable`1 source, Action`1 onNext)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClassaf.<<installPackageToAppDir>b__a8>d__b8.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.Utility.<LogIfThrows>d__54`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<ApplyReleases>d__78.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.UpdateManager.<FullInstall>d__f.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at Squirrel.Update.Program.<Install>d__37.MoveNext()
---> (Inner Exception #0) System.IO.IOException: Access to the path 'resources' is denied.
   at System.IO.DirectoryInfo.MoveTo(String destDirName)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClassaf.<>c__DisplayClassb4.<>c__DisplayClassb6.<installPackageToAppDir>b__ad(DirectoryInfo subdir)
   at Squirrel.Utility.<>c__DisplayClass18`1.<>c__DisplayClass1a.<ForEachAsync>b__17()
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.Utility.<>c__DisplayClass1e`1.<>c__DisplayClass20.<<ForEachAsync>b__1d>d__22.MoveNext()<---
<---

2016-01-11 22:07:25> Program: Starting Squirrel Updater: --install .
2016-01-11 22:07:25> Program: Starting install, writing to C:\Users\mypc\AppData\Local\SquirrelTemp
2016-01-11 22:07:25> Program: About to install to: C:\Users\mypc\AppData\Local\atom
2016-01-11 22:07:25> Program: Install path C:\Users\mypc\AppData\Local\atom already exists, burning it to the ground
2016-01-11 22:07:27> CheckForUpdateImpl: Failed to load local releases, starting from scratch: System.IO.DirectoryNotFoundException: Could not find a part of the path 'C:\Users\mypc\AppData\Local\atom\packages\RELEASES'.
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share)
   at Squirrel.Utility.LoadLocalReleases(String localReleaseFile)
   at Squirrel.UpdateManager.CheckForUpdateImpl.<CheckForUpdate>d__3d.MoveNext()
2016-01-11 22:07:27> CheckForUpdateImpl: Reading RELEASES file from C:\Users\mypc\AppData\Local\SquirrelTemp
2016-01-11 22:07:27> CheckForUpdateImpl: First run or local directory is corrupt, starting from scratch
2016-01-11 22:07:27> ApplyReleasesImpl: Writing files to app directory: C:\Users\mypc\AppData\Local\atom\app-1.3.3
2016-01-11 22:07:52> IEnableLogger: Failed to install package to app dir: System.AggregateException: One or more errors occurred. ---> System.IO.IOException: Access to the path 'resources' is denied.
   at System.IO.DirectoryInfo.MoveTo(String destDirName)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClassaf.<>c__DisplayClassb4.<>c__DisplayClassb6.<installPackageToAppDir>b__ad(DirectoryInfo subdir)
   at Squirrel.Utility.<>c__DisplayClass18`1.<>c__DisplayClass1a.<ForEachAsync>b__17()
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.Utility.<>c__DisplayClass1e`1.<>c__DisplayClass20.<<ForEachAsync>b__1d>d__22.MoveNext()
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClassaf.<>c__DisplayClassb4.<installPackageToAppDir>b__ac(DirectoryInfo ld)
   at Squirrel.EnumerableExtensions.ForEach[TSource](IEnumerable`1 source, Action`1 onNext)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClassaf.<<installPackageToAppDir>b__a8>d__b8.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.Utility.<LogIfThrows>d__54`1.MoveNext()
---> (Inner Exception #0) System.IO.IOException: Access to the path 'resources' is denied.
   at System.IO.DirectoryInfo.MoveTo(String destDirName)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClassaf.<>c__DisplayClassb4.<>c__DisplayClassb6.<installPackageToAppDir>b__ad(DirectoryInfo subdir)
   at Squirrel.Utility.<>c__DisplayClass18`1.<>c__DisplayClass1a.<ForEachAsync>b__17()
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.Utility.<>c__DisplayClass1e`1.<>c__DisplayClass20.<<ForEachAsync>b__1d>d__22.MoveNext()<---

2016-01-11 22:07:52> Unhandled exception: System.AggregateException: One or more errors occurred. ---> System.AggregateException: One or more errors occurred. ---> System.IO.IOException: Access to the path 'resources' is denied.
   at System.IO.DirectoryInfo.MoveTo(String destDirName)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClassaf.<>c__DisplayClassb4.<>c__DisplayClassb6.<installPackageToAppDir>b__ad(DirectoryInfo subdir)
   at Squirrel.Utility.<>c__DisplayClass18`1.<>c__DisplayClass1a.<ForEachAsync>b__17()
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.Utility.<>c__DisplayClass1e`1.<>c__DisplayClass20.<<ForEachAsync>b__1d>d__22.MoveNext()
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClassaf.<>c__DisplayClassb4.<installPackageToAppDir>b__ac(DirectoryInfo ld)
   at Squirrel.EnumerableExtensions.ForEach[TSource](IEnumerable`1 source, Action`1 onNext)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClassaf.<<installPackageToAppDir>b__a8>d__b8.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.Utility.<LogIfThrows>d__54`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<ApplyReleases>d__78.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.UpdateManager.<FullInstall>d__f.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at Squirrel.Update.Program.<Install>d__37.MoveNext()
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait()
   at Squirrel.Update.Program.executeCommandLine(String[] args)
   at Squirrel.Update.Program.main(String[] args)
---> (Inner Exception #0) System.AggregateException: One or more errors occurred. ---> System.IO.IOException: Access to the path 'resources' is denied.
   at System.IO.DirectoryInfo.MoveTo(String destDirName)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClassaf.<>c__DisplayClassb4.<>c__DisplayClassb6.<installPackageToAppDir>b__ad(DirectoryInfo subdir)
   at Squirrel.Utility.<>c__DisplayClass18`1.<>c__DisplayClass1a.<ForEachAsync>b__17()
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.Utility.<>c__DisplayClass1e`1.<>c__DisplayClass20.<<ForEachAsync>b__1d>d__22.MoveNext()
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClassaf.<>c__DisplayClassb4.<installPackageToAppDir>b__ac(DirectoryInfo ld)
   at Squirrel.EnumerableExtensions.ForEach[TSource](IEnumerable`1 source, Action`1 onNext)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClassaf.<<installPackageToAppDir>b__a8>d__b8.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.Utility.<LogIfThrows>d__54`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<ApplyReleases>d__78.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.UpdateManager.<FullInstall>d__f.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at Squirrel.Update.Program.<Install>d__37.MoveNext()
---> (Inner Exception #0) System.IO.IOException: Access to the path 'resources' is denied.
   at System.IO.DirectoryInfo.MoveTo(String destDirName)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClassaf.<>c__DisplayClassb4.<>c__DisplayClassb6.<installPackageToAppDir>b__ad(DirectoryInfo subdir)
   at Squirrel.Utility.<>c__DisplayClass18`1.<>c__DisplayClass1a.<ForEachAsync>b__17()
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.Utility.<>c__DisplayClass1e`1.<>c__DisplayClass20.<<ForEachAsync>b__1d>d__22.MoveNext()<---
<---

**Comment by 50Wliu** [comment]
/cc @paulcbetts 

Error is `System.IO.IOException: Access to the path 'resources' is denied.`
```
IEnableLogger: Failed to install package to app dir: System.AggregateException: One or more errors occurred. ---> System.IO.IOException: Access to the path 'resources' is denied.
at System.IO.DirectoryInfo.MoveTo(String destDirName)
```

**Comment by paulcbetts** [comment]
It seems that Avast has decided that it doesn't Like Electron-based apps, file a whitelist request

**Comment by xtoq** [comment]
@SHENKAR is this for Atom 1.3.3 or Atom 1.4.0?

**Comment by SHENKAR** [comment]
@xtoq I don't know. I just clicked the orange button on the main page!
@paulcbetts I'm using McAfee by the way.
---------------------------
Checksum information
---------------------------
Name: AtomSetup.exe
Size: 89570056 bytes (85 MB)

CRC32: 657EC7C0

CRC64: E450D5DBDAA6E0E9

SHA256: BB571569557AC04EEF1576815F9A36BD7A48BAB2B4560BCD441AF2A934062D23

SHA1: 8AF2A1AB00279B5F43C5B0A2E649C5DBFE4E45FE

BLAKE2sp: 2639653A2EC8F123451B8A8C3261372A88A9ED99CC2EF64E6D7FC972034328B1

---------------------------
OK   
---------------------------

**Comment by xtoq** [comment]
I believe that would be the stable version, 1.3.3.

@paulcbetts I can't seem to install the beta 1.4.0 alongside the stable 1.3.3, but I was able to upgrade from 1.3.x to the current 1.3.3 without issue. Should I open a separate ticket for the 1.4.0 version, or just tack onto this one? Wasn't sure since they are different codebases.

**Comment by Zireael07** [comment]
1.4.0 is no longer beta as of two days ago, just FYI.

**Comment by paulcbetts** [comment]
@xtoq Do you have the log file for when installing 1.4.0 failed?

**Comment by xtoq** [comment]
@paulcbetts No, when I click "view log" nothing happens, no log opens, no explorer opens. I can't find one generated with a simple Windows search; where would it live so I watch the dir while installing?

**Comment by paulcbetts** [comment]
@xtoq On initial install it's at `%LocalAppData%\SquirrelTemp`

**Comment by xtoq** [comment]
Thanks. I was able to view the log file there, and I uninstalled Atom completely and started over.

TLDR: I can't install both the stable and beta versions side by side on Windows 10; whichever is installed second fails to install and breaks the previous installation.


Steps to reproduce
-----

1. Download both the [stable](https://atom.io/download/windows) and [beta](https://atom.io/download/windows?channel=beta) versions for Windows.
2. Pick a version and install it.
3. Atom should open automatically after install; if not, open it to ensure it installed correctly.
3. Restart your computer.
4. Install the remaining version.


Expected outcome
-----

- You should be able to install and run both versions side-by-side as stated on the [website](https://atom.io/beta).


Actual outcome
-----

- Whichever version installed second will fail at setup, after roughly 5 seconds on the splash screen.
- ~~When you select "View Setup Log," the window closes but no other no windows are opened, no file is shown, no processes are started (from the process manager GUI), and no program starts.~~ This was a Windows 10 default viewer issue and not related to this install issue.
- When trying to run the previous installation, it now tells you that your system is missing node.dll. I suspect this is due to the cleanup processes that run with setup.


Additional information
-----

- Windows 10 professional 64-bit
- I have tried this with the newest versions available on the atom.io website: stable 1.4.0 and beta 1.5.0.

Error log (user name removed):

```
2016-01-15 17:11:13> Program: Starting Squirrel Updater: --install .
2016-01-15 17:11:13> Program: Starting install, writing to C:\Users\<username>\AppData\Local\SquirrelTemp
2016-01-15 17:11:13> Program: About to install to: C:\Users\<username>\AppData\Local\atom
2016-01-15 17:11:13> Program: Install path C:\Users\<username>\AppData\Local\atom already exists, burning it to the ground
2016-01-15 17:11:13> CheckForUpdateImpl: Failed to load local releases, starting from scratch: System.IO.DirectoryNotFoundException: Could not find a part of the path 'C:\Users\<username>\AppData\Local\atom\packages\RELEASES'.
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share)
   at Squirrel.Utility.LoadLocalReleases(String localReleaseFile)
   at Squirrel.UpdateManager.CheckForUpdateImpl.<CheckForUpdate>d__3d.MoveNext()
2016-01-15 17:11:13> CheckForUpdateImpl: Reading RELEASES file from C:\Users\<username>\AppData\Local\SquirrelTemp
2016-01-15 17:11:13> CheckForUpdateImpl: First run or local directory is corrupt, starting from scratch
2016-01-15 17:11:13> ApplyReleasesImpl: Writing files to app directory: C:\Users\<username>\AppData\Local\atom\app-1.5.0-beta0
2016-01-15 17:11:32> ApplyReleasesImpl: Moving file C:\Users\<username>\AppData\Local\atom\app-1.5.0-beta0\lib\net45\d3dcompiler_47.dll to C:\Users\<username>\AppData\Local\atom\app-1.5.0-beta0\d3dcompiler_47.dll
2016-01-15 17:11:32> ApplyReleasesImpl: Moving file C:\Users\<username>\AppData\Local\atom\app-1.5.0-beta0\lib\net45\atom.exe to C:\Users\<username>\AppData\Local\atom\app-1.5.0-beta0\atom.exe
2016-01-15 17:11:32> ApplyReleasesImpl: Moving file C:\Users\<username>\AppData\Local\atom\app-1.5.0-beta0\lib\net45\content_resources_200_percent.pak to C:\Users\<username>\AppData\Local\atom\app-1.5.0-beta0\content_resources_200_percent.pak
2016-01-15 17:11:32> ApplyReleasesImpl: Moving file C:\Users\<username>\AppData\Local\atom\app-1.5.0-beta0\lib\net45\content_shell.pak to C:\Users\<username>\AppData\Local\atom\app-1.5.0-beta0\content_shell.pak
2016-01-15 17:11:32> ApplyReleasesImpl: Moving file C:\Users\<username>\AppData\Local\atom\app-1.5.0-beta0\lib\net45\icudtl.dat to C:\Users\<username>\AppData\Local\atom\app-1.5.0-beta0\icudtl.dat
2016-01-15 17:11:32> ApplyReleasesImpl: Moving file C:\Users\<username>\AppData\Local\atom\app-1.5.0-beta0\lib\net45\libEGL.dll to C:\Users\<username>\AppData\Local\atom\app-1.5.0-beta0\libEGL.dll
2016-01-15 17:11:32> ApplyReleasesImpl: Moving file C:\Users\<username>\AppData\Local\atom\app-1.5.0-beta0\lib\net45\libGLESv2.dll to C:\Users\<username>\AppData\Local\atom\app-1.5.0-beta0\libGLESv2.dll
2016-01-15 17:11:32> ApplyReleasesImpl: Moving file C:\Users\<username>\AppData\Local\atom\app-1.5.0-beta0\lib\net45\LICENSE to C:\Users\<username>\AppData\Local\atom\app-1.5.0-beta0\LICENSE
2016-01-15 17:11:32> ApplyReleasesImpl: Moving file C:\Users\<username>\AppData\Local\atom\app-1.5.0-beta0\lib\net45\msvcp120.dll to C:\Users\<username>\AppData\Local\atom\app-1.5.0-beta0\msvcp120.dll
2016-01-15 17:11:32> ApplyReleasesImpl: Moving file C:\Users\<username>\AppData\Local\atom\app-1.5.0-beta0\lib\net45\msvcr120.dll to C:\Users\<username>\AppData\Local\atom\app-1.5.0-beta0\msvcr120.dll
2016-01-15 17:11:32> ApplyReleasesImpl: Moving file C:\Users\<username>\AppData\Local\atom\app-1.5.0-beta0\lib\net45\natives_blob.bin to C:\Users\<username>\AppData\Local\atom\app-1.5.0-beta0\natives_blob.bin
2016-01-15 17:11:32> ApplyReleasesImpl: Moving file C:\Users\<username>\AppData\Local\atom\app-1.5.0-beta0\lib\net45\node.dll to C:\Users\<username>\AppData\Local\atom\app-1.5.0-beta0\node.dll
2016-01-15 17:11:32> ApplyReleasesImpl: Moving file C:\Users\<username>\AppData\Local\atom\app-1.5.0-beta0\lib\net45\pdf.dll to C:\Users\<username>\AppData\Local\atom\app-1.5.0-beta0\pdf.dll
2016-01-15 17:11:32> ApplyReleasesImpl: Moving file C:\Users\<username>\AppData\Local\atom\app-1.5.0-beta0\lib\net45\vccorlib120.dll to C:\Users\<username>\AppData\Local\atom\app-1.5.0-beta0\vccorlib120.dll
2016-01-15 17:11:32> ApplyReleasesImpl: Moving file C:\Users\<username>\AppData\Local\atom\app-1.5.0-beta0\lib\net45\squirrel.exe to C:\Users\<username>\AppData\Local\atom\app-1.5.0-beta0\squirrel.exe
2016-01-15 17:11:32> ApplyReleasesImpl: Moving file C:\Users\<username>\AppData\Local\atom\app-1.5.0-beta0\lib\net45\snapshot_blob.bin to C:\Users\<username>\AppData\Local\atom\app-1.5.0-beta0\snapshot_blob.bin
2016-01-15 17:11:32> ApplyReleasesImpl: Moving file C:\Users\<username>\AppData\Local\atom\app-1.5.0-beta0\lib\net45\ui_resources_200_percent.pak to C:\Users\<username>\AppData\Local\atom\app-1.5.0-beta0\ui_resources_200_percent.pak
2016-01-15 17:11:32> ApplyReleasesImpl: Moving file C:\Users\<username>\AppData\Local\atom\app-1.5.0-beta0\lib\net45\xinput1_3.dll to C:\Users\<username>\AppData\Local\atom\app-1.5.0-beta0\xinput1_3.dll
2016-01-15 17:11:34> ApplyReleasesImpl: Squirrel Enabled Apps: [C:\Users\<username>\AppData\Local\atom\app-1.5.0-beta0\atom.exe]
2016-01-15 17:11:37> ApplyReleasesImpl: Starting fixPinnedExecutables
2016-01-15 17:11:37> ApplyReleasesImpl: fixPinnedExecutables: oldAppDirectories is empty, this is pointless
2016-01-15 17:11:37> ApplyReleasesImpl: Fixing up tray icons
2016-01-15 17:11:37> ApplyReleasesImpl: cleanDeadVersions: for version 1.5.0-beta0
2016-01-15 17:11:37> ApplyReleasesImpl: cleanDeadVersions: exclude folder app-1.5.0-beta0
```

**Comment by 50Wliu** [comment]
> TLDR: I can't install both the stable and beta versions side by side on Windows 10; whichever is installed second fails to install and breaks the previous installation.

Yeah, this is a known issue on Windows: I believe @maxbrunsfeld tried to fix it but ran into some nasty roadblocks along the way.

**Comment by xtoq** [comment]
Gah. It's not as big of a deal as it was when I first started using Atom before the update that improved load time on Windows, because I was trying to evaluate the improvements.

So is this the expected outcome of trying to install them both on the same system? If so, what's the reason that it can't just uninstall the currently installed version and then install the new version? Why does it seem to take 2 installs to get a different version? Windows is so fun! =\

**Comment by xtoq** [comment]
@Zireael07 Thanks, that's why I redid the test to include the new versions!

**Comment by paulcbetts** [comment]
> If so, what's the reason that it can't just uninstall the currently installed version and then install the new version?

I suspect that this is a bug in Atom's Squirrel hooks, the Slack app (also based on Squirrel) does exactly this - clicking on the setup.exe when it's already installed just results in the app being reinstalled.

**Comment by SHENKAR** [comment]
@paulcbetts I just found a solution to my problem. I disabled the Real-Time Scanning feature in McAfee during Atom installation and I had no problem. I am enjoying using it now. Thanks.

**Comment by 50Wliu** [comment]
@paulcbetts We've had a few more reports of the same error recently.

**Comment by noempujen** [comment]
Disable the Real-Time Scanning feature in McAfee before starting Atom installation. It works OK.

**Comment by ebiester** [comment]
Confirming that disabling real-time scanning in McAfee is causing the problem in another instance.

**Comment by paulcbetts** [comment]
@50Wliu I believe this is #10567 which is biting a lot of Windows apps right now (not just Squirrel or Electron-based ones)

**Comment by danielseltzer** [comment]
I'd like to thank this thread for solving a problem with failing Slack installs on Windows. Same error stack trace, no good information elsewhere. Disabled McAfee scan and re-installed successfully. Hopefully others will find this as well.

**Comment by 50Wliu** [comment]
This should be fixed in Atom 1.4.2.  Sorry for the inconvenience!

**Comment by JedBartlett** [comment]
Same exact stack trace
`Unhandled exception: System.AggregateException: One or more errors occurred. ---> System.AggregateException: One or more errors occurred. ---> System.IO.IOException: Access to the path 'resources' is denied.
   at System.IO.DirectoryInfo.MoveTo(String destDirName)`

@danielseltzer 's comment fixed it for me too:

> Disabled McAfee scan and re-installed successfully

The Version on the installer says 1.5.3, and is the same as the version listed under "Help > About Atom"  after disabling McAfee, so doesn't look like it was fixed in 1.4.2

**Comment by unseen1980** [comment]
That's a bit of an issue considering many of us working in corporate environments where disabling av is not an easy option due to active directory restrictions. Would be great if we find a workaround.

**Comment by agc93** [comment]
This is basically the same issue as #5773 from over a year ago. Is there no workaround for corporate users who can't turn off their AV?

**Comment by tesslerc** [comment]
Tried latest stable and beta, unable to install.
I also have a corporate av and unable to disable it in order to install.

**Comment by aravind-gk** [comment]
I faced the same problem. I temporarily disabled McAfee and that did the trick for me :)

**Comment by dkselw** [comment]
Same error with me on Windows 10 "Access to the path 'resources' is denied.".
The installation completed after disabling Kaspersky.

Atom version 1.8.0

**Comment by varunmalhotraminjar** [comment]
Download and tried to install the latest version on Windows 10. Same error : "Access to the path 'resources' is denied.".

**Comment by lee-dohm** [comment]
For everyone that is having problems installing Atom on Windows, please see the [Atom FAQ](https://discuss.atom.io/c/faq) entry:

https://discuss.atom.io/t/having-windows-installation-problems/26498

and follow the instructions there.

Locking the conversation here since this issue is no longer valid.
